superset-frontend/src/explore/components/controls/MetricsControl.jsx

```diff
@@ -194,6 +196,9 @@ export default class MetricsControl extends React.PureComponent {
   }
 
   onMetricEdit(changedMetric) {
+    changedMetric.label = changedMetric.hasCustomLabel
+      ? changedMetric.customLabel
+      : changedMetric.label;
```

Comment:
Can we change this so that we don't need to mutate `changedMetric`. Something like 
```typescript
const newMetric = { ...changedMetric }
newMetric.label = newMetric.hasCustomLabel ? newMetric.customLabel : newMetric.label
```
or similar? Optionally we could do this check in `onSave()` in `AdhocMetricEditPopover.jsx`, and reassign the label there to avoid having to rename the prop in `title`.

Reply:
Yeah, checking while saving would be better.